### ISSUE TITLE
Selinux preventing socket listen when running with systemd

### ISSUE THREAD

**Issue by runcom** [runcom/trust-plugin#5]
@rhatdan, I'm on F24 and getting:

```
sudo ausearch -m avc | grep trust
type=AVC msg=audit(1474376273.557:2075): avc:  denied  { listen } for  pid=1 comm="systemd" path="/run/docker/plugins/trust-plugin.sock" scontext=system_u:system_r:init_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=1
type=AVC msg=audit(1474376720.570:2252): avc:  denied  { listen } for  pid=1 comm="systemd" path="/run/docker/plugins/trust-plugin.sock" scontext=system_u:system_r:init_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=1
type=AVC msg=audit(1474464065.200:2703): avc:  denied  { listen } for  pid=1 comm="systemd" path="/run/docker/plugins/trust-plugin.sock" scontext=system_u:system_r:init_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=1
```

This is easily reproducible by:

```
$ make
$ sudo make install
$ sudo systemctl start trust-plugin.service
```

If instead I run the binary manually everything works fine:

```
$ sudo ./trust-plugin
[working]
```

**Comment by rhatdan** [comment]
Can you call it docker-trust-plugin and then run restorecon on it.

**Comment by rhatdan** [comment]
Although we probably should allow the AVC that you specified.

**Comment by runcom** [comment]
> Can you call it docker-trust-plugin and then run restorecon on it.

we can't call it that way I guess https://twitter.com/MacYET/status/775535642793086976

**Comment by runcom** [comment]
BTW, I'm trying what you suggested even if I prefer to have it called just `trust-plugin` and allow that AVC as you suggested also

**Comment by rhatdan** [comment]
Well that AVC would require a change to selinux-policy.  If we want to call it trust-plugin, then we need to change docker-selinux to label it as docker_exec_t, but we would need to do this for each plugin anyways, I was just going to label /usr/bin/container.\* as docker_exec_t, so that this would not happen with the next plugin.

**Comment by runcom** [comment]
> Well that AVC would require a change to selinux-policy. If we want to call it trust-plugin, then we need to change docker-selinux to label it as docker_exec_t, but we would need to do this for each plugin anyways, I was just going to label /usr/bin/container.\* as docker_exec_t, so that this would not happen with the next plugin.

right, repo and plugin changed to be "container-" prefixed - also Dan those binaries are usually under `/usr/libexec/docker/`

**Comment by runcom** [comment]
till we have a fix in selinux-policy  this makes selinux work again:

```
$ sudo chcon -t docker_exec_t /usr/libexec/docker/container-trust-plugin
```

/cc @ddarrah
